### PR TITLE
feat(perf-issues): use new auto-registered feature for consecutive-http

### PR DESCRIPTION
**Pull request by DominikB2014**
This PR makes two changes:

1. Allows the use of both `flag_disabled`  admin option and `problem-creation` system option in `is_creation_allowed_for_system`.
The new `flag_disabled` admin option is auto registered by the option feature handler in getsentry.
Plan is to eventually migrate all detectors to use the`flag_disabled` option, but going to test it out with one detector first!

2. Allows the use of auto-registered options in `getsentry` in our test suite.

TODO
- [x] Add tests

### CODE REVIEW

File: src/sentry/utils/performance_issues/base.py [hunk 1]
```diff
@@ -130,11 +131,17 @@ def is_creation_allowed_for_system(self) -> bool:
             return False
 
         try:
-            rate = options.get(system_option)
-        except options.UnknownOption:
-            rate = 0
+            creation_option_value: bool | float | None = options.get(system_option)
+            if isinstance(creation_option_value, bool):
+                return creation_option_value
```
Comment (DominikB2014) [inline src/sentry/utils/performance_issues/base.py:136]: I did use `options.lookup_key(key_name).type` as I couldn't figure out how to do the comparison
I tried 
```
type = options.lookup_key(key_name).type
type is bool
```
and
```
type = options.lookup_key(key_name).type
type == bool
```

File: src/sentry/utils/performance_issues/detectors/consecutive_http_detector.py [hunk 2]
```diff
@@ -136,4 +136,4 @@ def is_creation_allowed_for_organization(self, organization: Organization) -> bo
         )
 
     def is_creation_allowed_for_project(self, project: Project) -> bool:
-        return False
+        return True
```
Comment (DominikB2014) [inline src/sentry/utils/performance_issues/detectors/consecutive_http_detector.py:139]: Kinda confused why this is in the diff.
https://github.com/getsentry/sentry/pull/45780/files

File: src/sentry/options/defaults.py [hunk 3]
```diff
@@ -616,6 +616,7 @@
 register("performance.issues.compressed_assets.ea-rollout", default=0.0)
 register("performance.issues.compressed_assets.ga-rollout", default=0.0)
 register("performance.issues.consecutive_db.problem-creation", default=0.0)
+register("performance.issues.consecutive_http.flag_disabled", default=True)
```
[the remark this answers is not in the document]
Comment (DominikB2014) [reply src/sentry/options/defaults.py:619]: Looked into this more, discussed on Slack

File: src/sentry/testutils/helpers/options.py [hunk 4]
```diff
@@ -15,18 +17,27 @@ def override_options(options):
...
+        else:
+            return original_lookup(self, key)
+
     # Patch options into SENTRY_OPTIONS as well
     new_options = settings.SENTRY_OPTIONS.copy()
     new_options.update(options)
     with override_settings(SENTRY_OPTIONS=new_options):
         with patch.object(default_manager.store, "get", side_effect=new_get):
-            yield
+            with patch("sentry.options.OptionsManager.lookup_key", new=new_lookup):
```
Comment (k-fish) [inline src/sentry/testutils/helpers/options.py:42]: You don't need to nest these ctx managers you can actually just use `,`

**Review by k-fish — APPROVED** [review]
Nice debugging @DominikB2014 👍 Let's try this out

File: src/sentry/testutils/helpers/options.py (hunk 4 above)
Comment (DominikB2014) [reply src/sentry/testutils/helpers/options.py:42]: Good catch, will make the change

**Pull request merged**